### ISSUE TITLE
Node.js binding/npm package

### ISSUE THREAD

**Issue by Aalivexy** [carloskiki/pulldown-latex#11]
Currently, there seems to be no library in npm that converts latex to mathml core.

It would be great if we could provide Node.js binding or wasm-wasi to javascript binding!

I believe this can promote the popularity of mathml core.

**Comment by carloskiki** [comment]
Yes of course! This is one of the goals I have with this crate: having the library available as an `npm` package for `js`.

I do not have a lot of experience with publishing crates to `npm`, and I have never done it with a rust lib. If you have any idea on how it is done, it would gladly take the advice!

**Comment by kurabirko** [comment]
Hi, I recently found this project and agree that an npm package would be really nice to have. I would love to use pulldown as a remark or vite plugin for example. I think the [napi-rs](https://github.com/napi-rs/napi-rs) crate is widely used to bridge rust with node. I would like to take on this issue, however I don't really know all that much about rust honestly. If you don't have any plan to implement this yourself soon, I would like to give it a go anyway.

**Comment by carloskiki** [comment]
@kurabirko yes you should!

I think this is a great way to learn, I'll try to teach you as much as I can in my review of your PR.

**Comment by noClaps** [comment]
I was trying to work on this, and it turns out that you can't apply `#[wasm_bindgen]` to functions with lifetimes or type parameters, so generating WASM using `wasm-pack` isn't possible with the current implementation. Almost all of the exported functions and structs use lifetimes and/or type parameters, so adding WASM bindings won't be as simple.

https://users.rust-lang.org/t/wasm-bindgen-how-to-expose-existent-api-which-uses-lifetimes-to-js/71678

I'm not going to rewrite the code to remove lifetimes and type parameters from all of the exported functions and structs, but if @carloskiki or someone else is willing to do that then I'm happy to help add WASM support.

**Comment by gjtorikian** [comment]
I think https://github.com/bearcove/pulldown-latex-wasm does this now?

**Comment by carloskiki** [comment]
Looks like it, this could probably be added here.
